src/adapter/vertex.rs

```diff
@@ -36,31 +36,34 @@ pub enum VertexKind<'a> {
     ImplementedTrait(&'a Path, &'a Item),
     FunctionParameter(&'a str),
     FunctionAbi(&'a Abi),
+    Discriminant(String),
```

Comment:
```suggestion
    Discriminant(String),
```
I'd recommend either `Cow<'a, str>` or, even better, the borrowed-or-`Arc<str>` alternative. Vertices get cloned _a lot_, and we don't want to keep cloning the full `String` each time.